src/main/java/me/jellysquid/mods/sodium/client/util/workarounds/InGameChecks.java

```diff
@@ -0,0 +1,123 @@
+            "rendertype_translucent.vsh",
+            "rendertype_tripwire.vsh"
+    ));
+    private static final List<String> glslBlacklist = new ArrayList<>(Arrays.asList(
+            "light.glsl",
+            "fog.glsl"
+    ));
+
+    /**
+     * <a href="https://github.com/CaffeineMC/sodium-fabric/issues/1569">#1569</a>
+     * Iterate through all active resource packs, and detect resource packs which contain files matching the blacklist.
+     * An error message is shown for resource packs which replace terrain core shaders.
+     * A warning is shown for resource packs which modify the default light.glsl and fog.glsl shaders.
+     * Detailed information on shader files replaced or modified by resource packs is printed in the client log.
+     */
+    public static void checkIfCoreShaderLoaded() {
+        Collection<String> activeResourcePacks = MinecraftClient.getInstance().getResourcePackManager().getEnabledNames();
+        HashMap<String, MessageLevel> detectedResourcePacks = new HashMap<>();
+		
+        for (String s : activeResourcePacks) {
+			
+            if (!s.equals("vanilla") && !s.equals("fabric")) {
+                String resourcePackName = s.substring(5);
+				
+                try (ZipFile zip = new ZipFile(Paths.get(resourcePackDir + "\\" + resourcePackName).toFile())) {
+                    Enumeration<? extends ZipEntry> entries = zip.entries();
+
+                    while (entries.hasMoreElements()) {
+                        ZipEntry entry = entries.nextElement();
+                        String fileName = entry.getName().substring(entry.getName().lastIndexOf('/') + 1);
+						
+                        if (vshBlacklist.contains(fileName)) {
+							
+                            if (!detectedResourcePacks.containsKey(resourcePackName)) {
+                                detectedResourcePacks.put(resourcePackName, MessageLevel.SEVERE);
+                            } else if (detectedResourcePacks.get(resourcePackName) == MessageLevel.WARN) {
+                                detectedResourcePacks.replace(resourcePackName, MessageLevel.SEVERE);
+                            }
+							
+                            logMessageError("Resource pack '" + resourcePackName + "' replaces core shader '" + fileName + "'");
+                        }
+						
+                        if (glslBlacklist.contains(fileName)) {
+							
+                            if (!detectedResourcePacks.containsKey(resourcePackName)) {
+                                detectedResourcePacks.put(resourcePackName, MessageLevel.WARN);
+                            }
+							
+                            logMessageWarn("Resource pack '" + resourcePackName + "' modifies shader '" + fileName + "'");
+                        }
+                    }
+		
+                } catch (Exception e) {
+                    LOGGER.error("Could not read resource pack '" + resourcePackName + "'");
+                    e.printStackTrace();
```

Comment:
Is there a reason the exception can't be passed to the logger like this?:
```suggestion
                    LOGGER.error("Could not read resource pack '" + resourcePackName + "'", e);
```

Reply:
Nope, I think I accidentally left it like that when trying some stuff out. Good catch